devenv/lib/config.py

```diff
@@ -2,11 +2,97 @@
 
 import configparser
 import functools
+import os
 import sys
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import TypeAlias
 
+from devenv.constants import CI
 from devenv.constants import MACHINE
 
 
+@dataclass(frozen=True)
+class ConfigOpt:
+    name: str
+    prompt: str
+    formatter: Callable[[str], str] | None = None
+    default: Callable[[], str] = lambda: ""
+
+
+def _path_formatter(path: str) -> str:
+    return os.path.normpath(os.path.expanduser(path))
+
+
+CONFIG_OPTS = {
+    "coderoot": ConfigOpt(
+        "coderoot",
+        "# please enter the root directory you want to work in",
+        _path_formatter,
+        lambda: "~/code",
+    )
+}
+
+DEFAULT_CONFIG: Config = {"devenv": {"coderoot": "~/code"}}
+
+Config: TypeAlias = "dict[str, dict[str, str | None]]"
```

Comment:
Is the string literal necessary here?

Reply:
DEFAULT_CONFIG goes away, if that's what you're referring to. Otherwise, yes.